src/runtime_src/core/edge/user/hwctx_object.cpp

```diff
@@ -17,20 +19,39 @@ namespace zynqaie {
 	  , m_uuid(std::move(uuid))
 	  , m_slotidx(slotidx)
 	  , m_mode(mode)
-  {}
+  {
+#ifdef XRT_ENABLE_AIE
+    auto device{xrt_core::get_userpf_device(m_shim)};
+    m_aie_array = std::make_unique<Aie>(device);
+#endif
+  }
+
+#ifdef XRT_ENABLE_AIE
+  Aie*
+  hwctx_object::get_aie_array_from_hwctx()
+  {
+    return m_aie_array.get();
+  }
+
+  bool
+  hwctx_object::is_aie_registered()
+  {
+    return (nullptr != m_aie_array);
```

Comment:
Nit.  It is not custom C++ to compare a ptr against nullptr in a conditional.
```suggestion
    return m_aie_array;
```